spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -101,12 +105,26 @@ public Table getTable(StructType schema, Transform[] partitioning, Map<String, S
     SparkSession spark = SparkSession.active();
     setupDefaultSparkCatalog(spark);
     String path = options.get("path");
+
+    Long snapshotId = getPropertyAsLong(options, SparkReadOptions.SNAPSHOT_ID);
+    Long asOfTimestamp = getPropertyAsLong(options, SparkReadOptions.AS_OF_TIMESTAMP);
+    Preconditions.checkArgument(asOfTimestamp == null || snapshotId == null,
+        "Cannot specify both snapshot-id (%s) and as-of-timestamp (%s)", snapshotId, asOfTimestamp);
+    String selector = "";
```

Comment:
Instead of using an empty string, I think it is better to use `null` as a signalling value. Then you can use `!=` null when creating the path.

Reply:
Adopted your suggestion.